generator/src/main/java/com/scottlogic/deg/generator/restrictions/NoAllowedTypesRestriction.java

```diff
@@ -0,0 +1,35 @@
+package com.scottlogic.deg.generator.restrictions;
```

Comment:
It is a concrete type that expresses that there are no types allowed for the current field. e.g. the merge of `ofType string` and `not(ofType string)`.
This could be expressed as an empty list of allowed types using the `DataTypeRestrictions` type, i believe this simplifies the concept.
It is the complement of `AnyTypeRestriction`